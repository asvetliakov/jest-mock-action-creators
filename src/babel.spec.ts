import { transform } from "@babel/core";
import plugin from "./babel";

it("Doesn't do anything without mock call", () => {
    const source = `
        jest.mock("./e");
        import { a } from "b";
        import e from "./e";
        jest.mock("./b");
    `;

    expect(transform(source, { plugins: [plugin] }).code).toMatchSnapshot();
});

it("Creates mock expressions for imports", () => {
    const source = `
    import a from "./a";
    import { b, c } from "bc";
    import * as m from "module";
    import { default as e } from "../e";
    import { f as k } from "../../f";
    import * as mmm from "mmm";
    import x from "x";

    mockActionCreators(a, b, c, m, e, k, mmm.actions, x.actions);
    `;
    expect(transform(source, { plugins: [plugin] }).code).toMatchSnapshot();
});

it("Creates mock expressions for node require", () => {
    const source = `
    require("nonused");
    const a = require("a");
    const { b, c, d } = require("../bcd");

    mockActionCreators(a.actions, b, c, d);
    `;
    expect(transform(source, { plugins: [plugin] }).code).toMatchSnapshot();
});

it("Skips already mocked modules by jest", () => {
    const source = `
    jest.mock("ac1");
    import * as ac1 from "ac1";
    import { ab } from "ab";
    const anotherAc = require("../another");

    jest.doMock("../another");

    mockActionCreators(ac1.actions, ab, anotherAc.anotherActions);
    `;
    expect(transform(source, { plugins: [plugin] }).code).toMatchSnapshot();
});

it("Works with multiple mock calls", () => {
    const source = `
    import { a } from "a";
    import * as b from "b";

    mockActionCreators(a);
    mockActionCreators(b.actions);
    `;
    expect(transform(source, { plugins: [plugin] }).code).toMatchSnapshot();
});

it("Don't create mocks for already processed modules", () => {
    const source = `
    import { a } from "a";
    import * as b from "b";
    import c from "c";

    mockActionCreators(a, b.actions, a, b.actions);
    mockActionCreators(c, a);
    `;
    expect(transform(source, { plugins: [plugin] }).code).toMatchSnapshot();
});

it("Works when mockActionCreators is being renamed in import", () => {
    const source = `
    import { mockActionCreators as mac } from "a";
    import * as b from "b";
    import c from "c";

    mac(b.actions, c);
    `;
    expect(transform(source, { plugins: [plugin] }).code).toMatchSnapshot();
});

it("Inserts mocks after use strict", () => {
    const source = `
    "use strict";

    import { a } from "a";
    import * as b from "b";

    mockActionCreators(a, b);
    `;
    expect(transform(source, { plugins: [plugin] }).code).toMatchSnapshot();
});