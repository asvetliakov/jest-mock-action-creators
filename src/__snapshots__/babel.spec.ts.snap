// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates mock expressions for imports 1`] = `
"jest.mock(\\"x\\");
jest.mock(\\"mmm\\");
jest.mock(\\"../../f\\");
jest.mock(\\"../e\\");
jest.mock(\\"module\\");
jest.mock(\\"bc\\");
jest.mock(\\"./a\\");
import a from \\"./a\\";
import { b, c } from \\"bc\\";
import * as m from \\"module\\";
import { default as e } from \\"../e\\";
import { f as k } from \\"../../f\\";
import * as mmm from \\"mmm\\";
import x from \\"x\\";
mockActionCreators(a, b, c, m, e, k, mmm.actions, x.actions);"
`;

exports[`Creates mock expressions for node require 1`] = `
"jest.mock(\\"../bcd\\");
jest.mock(\\"a\\");

require(\\"nonused\\");

const a = require(\\"a\\");

const {
  b,
  c,
  d
} = require(\\"../bcd\\");

mockActionCreators(a.actions, b, c, d);"
`;

exports[`Doesn't do anything without mock call 1`] = `
"jest.mock(\\"./e\\");
import { a } from \\"b\\";
import e from \\"./e\\";
jest.mock(\\"./b\\");"
`;

exports[`Don't create mocks for already processed modules 1`] = `
"jest.mock(\\"c\\");
jest.mock(\\"b\\");
jest.mock(\\"a\\");
import { a } from \\"a\\";
import * as b from \\"b\\";
import c from \\"c\\";
mockActionCreators(a, b.actions, a, b.actions);
mockActionCreators(c, a);"
`;

exports[`Inserts mocks after use strict 1`] = `
"\\"use strict\\";

jest.mock(\\"b\\");
jest.mock(\\"a\\");
import { a } from \\"a\\";
import * as b from \\"b\\";
mockActionCreators(a, b);"
`;

exports[`Skips already mocked modules by jest 1`] = `
"jest.mock(\\"ab\\");
jest.mock(\\"ac1\\");
import * as ac1 from \\"ac1\\";
import { ab } from \\"ab\\";

const anotherAc = require(\\"../another\\");

jest.doMock(\\"../another\\");
mockActionCreators(ac1.actions, ab, anotherAc.anotherActions);"
`;

exports[`Works when mockActionCreators is being renamed in import 1`] = `
"jest.mock(\\"c\\");
jest.mock(\\"b\\");
import { mockActionCreators as mac } from \\"a\\";
import * as b from \\"b\\";
import c from \\"c\\";
mac(b.actions, c);"
`;

exports[`Works with multiple mock calls 1`] = `
"jest.mock(\\"b\\");
jest.mock(\\"a\\");
import { a } from \\"a\\";
import * as b from \\"b\\";
mockActionCreators(a);
mockActionCreators(b.actions);"
`;
